java/gradle.java/src/org/netbeans/modules/gradle/java/JavaRuntimeManagerImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.gradle.java;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import org.netbeans.api.java.platform.JavaPlatform;
+import org.netbeans.api.java.platform.JavaPlatformManager;
+import org.netbeans.api.java.platform.PlatformsCustomizer;
+import org.netbeans.modules.gradle.spi.execute.JavaRuntimeManager;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+@ServiceProvider(service = JavaRuntimeManager.class, position = 0)
+public class JavaRuntimeManagerImpl implements JavaRuntimeManager {
+    private static final String PROP_PLATFORM_ID = "platform.ant.name"; //NOI18N
+
+    @Override
+    public Set<JavaRuntime> getAvailableRuntimes() {
+        Set<JavaRuntime> ret = new TreeSet<>();
+        JavaPlatformManager jpm = JavaPlatformManager.getDefault();
```

Comment:
wouldn't injecting JavaPlatformMaganer through constructor be more flexible?

Reply:
Not necessary. And what do you mean on `injecting` here?

Reply:
There is no JavaRuntimeManagerImpl in the final PR. That was just good enough to get the things going. Later on it was replaced by JavaRuntimePlatformBridge